core/src/main/java/com/microsoft/applicationinsights/internal/channel/common/TransmissionFileSystemOutput.java

```diff
@@ -92,12 +98,17 @@
     private final ArrayList<File> cacheOfOldestFiles = new ArrayList<File>();
     private final HashSet<String> filesThatAreBeingLoaded = new HashSet<String>();
 
-    public TransmissionFileSystemOutput() {
-        this(new File(System.getProperty("java.io.tmpdir"), TRANSMISSION_DEFAULT_FOLDER).getPath());
-    }
+    public TransmissionFileSystemOutput(String folderPath, String maxTransmissionStorageCapacity) {
+        if (folderPath == null) {
+            folderPath = new File(System.getProperty("java.io.tmpdir"), TRANSMISSION_DEFAULT_FOLDER).getPath();
+        }
 
-    public TransmissionFileSystemOutput(String folderPath) {
-        Preconditions.checkNotNull(folderPath, "folderPath must be a non-null value");
+        capacityEnforcer = LimitsEnforcer.createWithClosestLimitOnError(MIN_CAPACITY_MEGABYTES,
+                                                                        MAX_CAPACITY_MEGABYTES,
+                                                                        DEFAULT_CAPACITY_MEGABYTES,
+                                                                        MAX_TRANSMISSION_STORAGE_CAPACITY_NAME,
+                                                                        maxTransmissionStorageCapacity);
+        capacityInKB = capacityEnforcer.getCurrentValue() * 1024;
```

Comment:
Since you're using 1024 in multiple places please define it as a constant (KILO?)